openedx/edx-enterprise
# feat: creating new enterprise group members endpoint

Adding in an endpoint for the data table that is created in this ticket. We will be fetching info about the user (email, joined_org, name) from the enterprise_customer_user, and then fetching the total number of enrollments associated with that enterprise user. 

[Jira](https://2u-internal.atlassian.net/browse/ENT-9670)

Members are sorted on name
![Screenshot 2024-11-19 at 10 47 17 PM](https://github.com/user-attachments/assets/616b7ae3-5a8a-4401-8191-023ef34e74b0)

Testing: 
The link is http://localhost:18000/enterprise/api/v1/enterprise-customer-members/{enterprise-uuid}
- Check that each row has an enterprise_customer_user dict with info like first/last name and an enrollment count
- Add `?user_query=test` to url to make sure it is only filtering for either emails or names that contain the user query

**Merge checklist:**
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.

**Post merge:**
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
    - Trigger the '[Upgrade one Python dependency](https://github.com/openedx/edx-platform/actions/workflows/upgrade-one-python-dependency.yml)' action against master in edx-platform with new version number to generate version bump PR
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.
